test/bin/scenario.sh

```diff
@@ -479,12 +478,18 @@ launch_vm() {
             graphics_args="vnc,listen=0.0.0.0"
         fi
 
+        # Make sure the virt-install command times out after a predefined period.
+        # The 'timeout' command sends the HUP signal and, if the process does not
+        # exit after 1m, it sends the KILL signal to terminate the process.
+        # Note: Using the '--wait <time>' virt-install options may not work for
+        # failed installations when 'unbuffer' command is used.
+        local timeout_install="timeout -v --kill-after=1m ${VM_BOOT_TIMEOUT}s"
         # When bash creates a background job (using `&`),
         # the bg job does not get its own TTY.
         # If the TTY is not provided, virt-install refuses
         # to attach to the console. `unbuffer` provides the TTY.
         # shellcheck disable=SC2086
-        if ! unbuffer sudo virt-install \
+        if ! ${timeout_install} unbuffer sudo virt-install \
```

Comment:
How many different places are we applying timeouts in the call stack by the time we reach here? Does the call to parallel in the phase script also have a timeout, for example?

Reply:
We have taken a similar approach when running tests. It allows for per-test override of timeouts - so, if necessary, we can also allow per-test VM boot timeouts here in the future.

Reply:
OK, as long as we're consistent in only applying timeouts at the "bottom" of the call stack, this change makes sense.